lib/daryl_inject.rb

```diff
@@ -0,0 +1,15 @@
+class Array
+
+  def daryl_inject(result = self.first, &block)
+    return result if self.length == 1
+    self.shift if result == self.first
+    self.each { |x| result = block.call(result, x) }
+    result
+  end
+
+  def daryl_inject2(result = self.first, &block)
```

Comment:
Unused method argument - `result`. If it's necessary, use `_` or `_result` as an argument name to indicate that it won't be used.